tasks/contractAddressManagement.js

```diff
@@ -0,0 +1,37 @@
+const { task } = require("hardhat/config");
+const fs = require('fs');
+const contracts = require("../contractAddresses.json");
+const path = require("path");
+const deploymentsDirectory = "../deployments";
+
+task("exportAddress", "exsport contract addresses to contractAddresses.json file")
```

Comment:
tbh using a task and scanning the deployments folder seems a bit over-engineered to me. Do we ever need to use the CLI to run this task manually, or does it always need to be executed when we do a new deployment? To me it seems as if the latter would be the case. 

If this is true, I would suggest to do the following: We only need the address of a deployed contract and the name of the network. We have access to both information in the `deployFunction`. One approach that I would find a little bit more straightforward would be to in each `deployFunction`:

1. load and parse the json file -> get contractAddresses object
2. grab the network name (see [example](https://github.com/PrimeDAO/contracts-v2/blob/chore/hardhat-deploy-and-tasks/deploy/01_deploy_migration_contracts.js#L10) where we use ther network name in deployFunction)
3. after the deployment of a contract grab the contract address (see [example](https://github.com/fabianschu/clone_factory/blob/main/deploy/00_deploy_contracts.ts#L11) from another repo)
4. use network name and contract address to modify the object
5. save object to filesystem

Reply:
sure we can do that too, for each deployment, we add the contract address to the contractAddresses.json. right? And then we remove the export address task.

Reply:
If with "add the contract address" you mean "add contract address or replace if contract address for network already exists" (e.g. if we deploy multiple times to rinkeby to test stuff), then yes.

I would prefer this approach since it seems a bit more straightforward to me, but it's up to you :)